src/main/java/org/springframework/data/redis/core/StreamOperations.java

```diff
@@ -281,6 +281,22 @@ default PendingMessages pending(K key, Consumer consumer) {
 	 */
 	PendingMessages pending(K key, String group, Range<?> range, long count);
 
+	/**
+	 * Obtain detailed information about pending {@link PendingMessage messages} for a given {@link Range} within a
+	 * {@literal consumer group}.
+	 *
+	 * @param key the {@literal key} the stream is stored at. Must not be {@literal null}.
+	 * @param group the name of the {@literal consumer group}. Must not be {@literal null}.
+	 * @param range the range of messages ids to search within. Must not be {@literal null}.
+	 * @param count limit the number of results.
+	 * @param idleMilliSeconds the time limit until the pending messages are idle. Must not be {@literal null}.
+	 * @return pending messages for the given {@literal consumer group} or {@literal null} when used in pipeline /
+	 *         transaction.
+	 * @see <a href="https://redis.io/commands/xpending">Redis Documentation: xpending</a>
+	 * @since 2.3
+	 */
+	PendingMessages pending(K key, String group, Range<?> range, long count, long idleMilliSeconds);
```

Comment:
For both added methods (`pending`), it would make sense to accept `idleTime` as `long` and have a `TimeUnit` to avoid potential conversion bugs in calling code.